MQTTPacket/test/test2.c

```diff
@@ -533,10 +533,10 @@ int test3(struct Options options)
 
 	MQTTProperty one;
 	one.identifier = USER_PROPERTY;
-	one.value.data.data = "user property name";
-	one.value.data.len = strlen(one.value.data.data) + 1;
-	one.value.value.data = "user property value";
-	one.value.value.len = strlen(one.value.value.data) + 1;
+	one.value.string_pair.key.data = "user property name";
+	one.value.string_pair.key.len = strlen(one.value.string_pair.key.data) + 1;
```

Comment:
I'm not sure about the +1 here. This would just add the `\0` character which, if I correctly interpreted the [spec](https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901027), shouldn't be present in the string:

> A UTF-8 Encoded String MUST NOT include an encoding of the null character U+0000. [MQTT-1.5.4-2]

Reply:
Yes this is correct. The test fails because of this when run against the broker in paho.mqtt.testing.

Reply:
I've fixed this in #241 and cherry-picked the commit from here (ECA already signed by @MrEthan according to the ECA bot below).
#241 supersedes this PR.